### ISSUE TITLE
Allow connector services to be intercepted and augmented by extensions

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3231]
When a connector is invoking its service, extensions (e.g. SSL) should have the option of intercepting that invocation and doing custom things. For example, the SSL Manager should have the ability to use the specified custom SSL context when testing an outbound connection from the Administrator.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3328
Reporter: narupley
Created: 2014-06-09T14:12:10.000-0700

**Comment by rbeckman-nextgen** [comment]
Revision 7188: Added new plugin point for connector service extensions. These have the ability to intercept connector service invocations and inject their own logic if need be. On the client side, connector properties plugins also have the ability to intercept the response returned from connector services and perform whatever UI actions they need to. The HTTP and Web Service connectors have been modified to use the new client framework for this, and implement the handleConnectorServiceResponse method.

Imported Comment. Original Details:
Author: narupley
Created: 2014-06-09T15:24:44.000-0700

**Comment by rbeckman-nextgen** [comment]
Created several channels in 303 using the old ssl manager. Configured these channels to be able to send messages between them. I imported these into 3.1, selecting yes to migrate, and reconfigured them to work with the new SSL Manager. Verified that all the SSL specific settings work as intended whether it be from migration, configuring a new channel or cloning an existing channel.

To ensure no channels broke, I also tested the channels after uninstalling the SSL extension, or installing it where it wasn't present. For good measure, I made the old 303 SSL build talk to channels in 3.1 using the SSL Manager.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-09-17T14:03:27.000-0700
